Case Study on Weather report between 2006 - 2010

In [ ]:
pip install pandas

In [ ]:
import pandas as pd

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
df = pd.read_csv('/content/drive/MyDrive/weatherHistory.csv')
df

Formatted Date  ...                           Daily Summary
0      2006-04-01 00:00:00.000 +0200  ...       Partly cloudy throughout the day.
1      2006-04-01 01:00:00.000 +0200  ...       Partly cloudy throughout the day.
2      2006-04-01 02:00:00.000 +0200  ...       Partly cloudy throughout the day.
3      2006-04-01 03:00:00.000 +0200  ...       Partly cloudy throughout the day.
4      2006-04-01 04:00:00.000 +0200  ...       Partly cloudy throughout the day.
...                              ...  ...                                     ...
96448  2016-09-09 19:00:00.000 +0200  ...  Partly cloudy starting in the morning.
96449  2016-09-09 20:00:00.000 +0200  ...  Partly cloudy starting in the morning.
96450  2016-09-09 21:00:00.000 +0200  ...  Partly cloudy starting in the morning.
96451  2016-09-09 22:00:00.000 +0200  ...  Partly cloudy starting in the morning.
96452  2016-09-09 23:00:00.000 +0200  ...  Partly cloudy starting in the morning.

[96453 rows x 12 columns]

In [ ]:
maxTem = df['Temperature (C)'].max()
maxTem

39.90555556

In [ ]:
minTem = df['Temperature (C)'].min()
minTem

-21.82222222

In [ ]:
df['Temperature (C)'].mean()

11.932678437519867

In [ ]:
df['Formatted Date'][df['Temperature (C)']== maxTem]

12759    2007-07-22 15:00:00.000 +0200
Name: Formatted Date, dtype: object

In [ ]:
df['Formatted Date'][df['Temperature (C)'] == minTem]

54847    2012-02-10 07:00:00.000 +0100
Name: Formatted Date, dtype: object

In [ ]:
maxwin = df['Wind Speed (km/h)'].max()
maxwin

63.8526

In [ ]:
minwin = df['Wind Speed (km/h)'].min()
minwin 

0.0

In [ ]:
df['Formatted Date'][df['Wind Speed (km/h)'] == maxwin]

12181    2007-01-29 13:00:00.000 +0100
Name: Formatted Date, dtype: object

In [ ]:
df['Wind Speed (km/h)'].mean()

10.810640140793208

In [32]:
df['Summary'].max()

'Windy and Partly Cloudy'

In [33]:
df['Daily Summary'].max()

'Windy in the afternoon.'

In [37]:
df['Humidity'].max()

1.0

In [38]:
df['Humidity'].min()

0.0

In [36]:
df['Humidity'].mean()

0.7348989663358906

In [ ]:
OverallComparison = df.describe(include='all')
OverallComparison

Formatted Date  ...                      Daily Summary
count                           96453  ...                              96453
unique                          96429  ...                                214
top     2010-08-02 12:00:00.000 +0200  ...  Mostly cloudy throughout the day.
freq                                2  ...                              20085
mean                              NaN  ...                                NaN
std                               NaN  ...                                NaN
min                               NaN  ...                                NaN
25%                               NaN  ...                                NaN
50%                               NaN  ...                                NaN
75%                               NaN  ...                                NaN
max                               NaN  ...                                NaN

[11 rows x 12 columns]

The maximum Temperature recorded in this time period is 39.90555556 C on 2007-07-22

The minimum Temperature recorded in this time period is -21.82222222 C on 2012-02-10

The average Temperature recorded in this time interval is 11.932678437519867 C

The maximum WindSpeed recorded in this time period is 63.8526 km/h on 2007-01-29

The minimum windspeed recorded in this time period is 0.0 km/h

The average windspeed recorded in this time period is 10.810640140793208 km/h


In this time period the weather was mostly Windy and Partly Cloudy

In this time period the average weather on a Daily Basis was  Windy in the afternoon

The maximum Humidity recorded in this time period is 1.0

The minimum Humidity recorded in this time period is 0.0

The average Humitdity Recorded in this time period is 0.7348989663358906